### PR TITLE
test(covenant): improve test coverage

### PR DESCRIPTION
**Pull request by sdbondi**
Description
---
- adds more tests for covenant modules
- fixes typo in `CovenantReadExt` trait
- returns error on invalid `Option<T>` types in `is_eq`

Motivation and Context
---
Previous coverage:
image

Coverage is now:
PENDING

How Has This Been Tested?
---
Tests pass

### CODE REVIEW

File: base_layer/core/src/covenants/decoder.rs [hunk 1]
```diff
@@ -127,13 +127,41 @@ mod test {
...
+            fields::OutputField,
+            filters::CovenantFilter,
+        },
     };
 
     #[test]
     fn it_immediately_ends_iterator_given_empty_bytes() {
         let buf = &[] as &[u8; 0];
         assert!(CovenantTokenDecoder::new(&mut &buf[..]).next().is_none());
+        assert!(CovenantTokenDecoder::new(&mut &buf[..]).next().is_none());
```
Comment (SWvheerden) [inline base_layer/core/src/covenants/decoder.rs]: Why do this twice?

**Review by stringhandler — DISMISSED** [review]
utACK

**Comment by SWvheerden** [comment]
This looks good, 
There are a few naming refactors as well as the new testing code in the PR. 
Spent about 20 minutes reviewing this to ensure that all is well. 
This PR looks good:
But the test fails locally and in ci: `covenants::fields::test::output_field::is_eq::it_returns_false_if_not_eq `

File: base_layer/core/src/covenants/decoder.rs (hunk 1 above)
Comment (sdbondi) [reply base_layer/core/src/covenants/decoder.rs]: Checks the iterator behaviour, that it continues to return None - not necessary, removed

**Review by SWvheerden — APPROVED** [review]
LGTM

**Review by stringhandler — APPROVED** [review]

**Pull request merged**